pkg/lib/html2po.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python3
+# This file is part of Cockpit.
+#
+# Copyright (C) 2023 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+# Extracts translatable strings from HTML files in the following forms:
+#
+# <tag translate>String</tag>
+# <tag translate context="value">String</tag>
+# <tag translate="...">String</tag>
+# <tag translate-attr attr="String"></tag>
+#
+# Supports the following angular-gettext compatible forms:
+#
+# <translate>String</translate>
+# <tag translate-plural="Plural">Singular</tag>
+#
+# Note that some of the use of the translated may not support all the strings
+# depending on the code actually using these strings to translate the HTML.
+
+import argparse
+import os
+import sys
+from typing import Dict
+
+import bs4
```

Comment:
I'm not a fan of introducing a new parser dependency to be honest. Our HTML should be regular enough for ElementTree to parse, and if not, let's fix that. XML is good!

Reply:
> Our HTML should be regular enough for ElementTree to parse, and if not, let's fix that. XML is good!

Erm, no, HTML is not XML -- examples:
- ```
  <html ng-csp>
  <title translate>Software updates</title>
  ```
  value missing attributes
- ```
  <meta charset="utf-8">
  ```
  missing end tag

I understand the willing to not add a new dependency, however for this is definitely needed (since you cannot do assumption on HTML).

Reply:
Yes, that's what I meant --- the `ng-csp` tag is useless and can be dropped (we haven't used angular in years), `<meta>` can be closed properly with `/>`, and it should be `translate="yes"` (same with `hidden="yes"`). That would obviously be a separate PR (I'm happy to help with that)

Reply:
I couldn't resist and filed PR #18837 for doing that. This actually caught some actual bugs!